Makefile

```diff
@@ -264,6 +268,9 @@ ifeq ($(THREAD_MODEL), posix)
 # Specify the tls-model until LLVM 15 is released (which should contain
 # https://reviews.llvm.org/D130053).
 CFLAGS += -mthread-model posix -pthread -ftls-model=local-exec
+
+# Include cloudlib's directory to access the structure definition of clockid_t
+CFLAGS += -I$(LIBC_BOTTOM_HALF_CLOUDLIBC_SRC)
```

Comment:
For `pthread_condattr_setclock` (in top half) we need a definition of the `clockid_t` struct, which is defined in bottom half. Definition of `clockid_t` at least for now is not required to be a part of public interface, therefore I don't think we have to make it public at that stage. At the same time, I'm a bit worried that adding the whole cloudlib source dir as include by default to top half will lead to some weird usage if we're not careful enough. Another option would be to have a dedicated folder with headers shared between bottom and top half (but not available in public interface).

I'm open for suggestions.

Reply:
This solution seems ok to me.

Reply:
ok, let's keep it as it is for now then. Thanks for the feedback.